sympy/sets/conditionset.py

```diff
@@ -34,6 +36,15 @@ def __new__(cls, sym, condition, base_set):
             return S.EmptySet
         if condition == S.true:
             return base_set
+        if isinstance(base_set, EmptySet):
+            return base_set
+        if isinstance(base_set, FiniteSet):
```

Comment:
Looks great. As long as tests don't see something that we don't, this is ready to go. +1